src/py/flwr/client/message_handler/message_handler.py

```diff
@@ -93,7 +92,12 @@ def handle_legacy_message_from_msgtype(
     client_fn: ClientFn, message: Message, context: Context
 ) -> Message:
     """Handle legacy message in the inner most mod."""
-    client = client_fn(str(context.partition_id))
+    try:
+        client = client_fn(message.metadata.dst_node_id, context.partition_id)
+    except Exception as ex:  # pylint: disable=broad-exception-caught
+        log(DEBUG, ex)
+        # Attempt execution `client_fn` as it was done before flwr 1.10.0
+        client = client_fn(str(context.partition_id))  # type: ignore
```

Comment:
This should not happen in the `except` block b/c the body of a `client_fn` could raise an exception that is unrelated to the signature of `client_fn`. Instead of this, we should inspect the signature of the `client_fn` and call it with either one or two arguments (and no retry in `except`).

Reply:
right

Reply:
resolved in #3700